Sources/ShopifyCheckoutKit/AnalyticsEvents.swift

```diff
@@ -0,0 +1,4594 @@
+    }
+
+    func jsonString(encoding: String.Encoding = .utf8) throws -> String? {
+        return String(data: try self.jsonData(), encoding: encoding)
+    }
+}
+
+// MARK: - PixelEventsSearchSubmittedData
+struct PixelEventsSearchSubmittedData: Codable {
+    let searchResult: SearchResult?
+}
+
+// MARK: PixelEventsSearchSubmittedData convenience initializers and mutators
+
+extension PixelEventsSearchSubmittedData {
+    init(data: Data) throws {
+        self = try newJSONDecoder().decode(PixelEventsSearchSubmittedData.self, from: data)
+    }
+
+    init(_ json: String, using encoding: String.Encoding = .utf8) throws {
+        guard let data = json.data(using: encoding) else {
+            throw NSError(domain: "JSONDecoding", code: 0, userInfo: nil)
+        }
+        try self.init(data: data)
+    }
+
+    init(fromURL url: URL) throws {
+        try self.init(data: try Data(contentsOf: url))
+    }
+
+    func with(
+        searchResult: SearchResult?? = nil
+    ) -> PixelEventsSearchSubmittedData {
+        return PixelEventsSearchSubmittedData(
+            searchResult: searchResult ?? self.searchResult
+        )
+    }
+
+    func jsonData() throws -> Data {
+        return try newJSONEncoder().encode(self)
+    }
+
+    func jsonString(encoding: String.Encoding = .utf8) throws -> String? {
+        return String(data: try self.jsonData(), encoding: encoding)
+    }
+}
+
+/// An object that contains the metadata of when a search has been performed.
+// MARK: - SearchResult
+struct SearchResult: Codable {
+    let productVariants: [ProductVariant]?
+    /// The search query that was executed
+    let query: String?
+}
+
+// MARK: SearchResult convenience initializers and mutators
+
+extension SearchResult {
+    init(data: Data) throws {
+        self = try newJSONDecoder().decode(SearchResult.self, from: data)
+    }
+
+    init(_ json: String, using encoding: String.Encoding = .utf8) throws {
+        guard let data = json.data(using: encoding) else {
+            throw NSError(domain: "JSONDecoding", code: 0, userInfo: nil)
+        }
+        try self.init(data: data)
+    }
+
+    init(fromURL url: URL) throws {
+        try self.init(data: try Data(contentsOf: url))
+    }
+
+    func with(
+        productVariants: [ProductVariant]?? = nil,
+        query: String?? = nil
+    ) -> SearchResult {
+        return SearchResult(
+            productVariants: productVariants ?? self.productVariants,
+            query: query ?? self.query
+        )
+    }
+
+    func jsonData() throws -> Data {
+        return try newJSONEncoder().encode(self)
+    }
+
+    func jsonString(encoding: String.Encoding = .utf8) throws -> String? {
+        return String(data: try self.jsonData(), encoding: encoding)
+    }
+}
+
+/// The name of the customer event
+enum PixelEventsSearchSubmittedName: String, Codable {
+    case searchSubmitted = "search_submitted"
+}
+
+// MARK: - Browser
+struct Browser: Codable {
```

Comment:
I think this is unrelated to events and can be dropped, along with the types it references `BrowserCookie`, `SendBeacon` etc)